src/wallet/interfaces.cpp

```diff
@@ -458,6 +458,11 @@ class WalletImpl : public Wallet
     bool canGetAddresses() override { return m_wallet->CanGetAddresses(); }
     bool hasExternalSigner() override { return m_wallet->IsWalletFlagSet(WALLET_FLAG_EXTERNAL_SIGNER); }
     bool privateKeysDisabled() override { return m_wallet->IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS); }
+    bool taprootEnabled() override {
+        if (m_wallet->IsLegacy()) return false;
```

Comment:
Is this special case needed? Wouldn't it return nullptr below?

Reply:
Maybe not, but since legacy wallets don't have descriptors, it does seem cleaner.